Gordon360/Services/RecIM/ActivityService.cs

```diff
@@ -0,0 +1,146 @@
+                                Series = a.Series
+                                        .Select(s => new SeriesViewModel
+                                        {
+                                            ID = s.ID,
+                                            Name = s.Name,
+                                            StartDate = s.StartDate,
+                                            EndDate = s.EndDate,
+                                            Type = _context.SeriesType
+                                                    .FirstOrDefault(st => st.ID == s.TypeID)
+                                                    .Description
+                                        }).ToList()
+                            });
+            return activities;
+        }
+        public IEnumerable<ActivityViewModel> GetActivitiesByTime(DateTime? time)
+        {
+            if (time is null)
+            {
+                return GetActivities().Where(a => !a.Completed);
+            }
+            else
+            {
+                return GetActivities().Where(a => a.RegistrationEnd > time);
+            }
+        }
+        public ActivityViewModel? GetActivityByID(int activityID)
+        {
+            var activity = _context.Activity.Where(a => a.ID == activityID)
+                            .Select(a => new ActivityViewModel
+                            {
+                                ID = a.ID,
+                                Name = a.Name,
+                                RegistrationStart = a.RegistrationStart,
+                                RegistrationEnd = a.RegistrationEnd,
+                                Sport = _context.Sport
+                                        .FirstOrDefault(s => s.ID == a.SportID),
+                                Status = _context.ActivityStatus
+                                        .FirstOrDefault(s => s.ID == a.StatusID)
+                                        .Description,
+                                MinCapacity = a.MinCapacity,
+                                MaxCapacity = a.MaxCapacity,
+                                SoloRegistration = a.SoloRegistration,
+                                Logo = a.Logo,
+                                Completed = a.Completed,
+                                Series = _seriesService.GetSeriesByActivityID(a.ID),
+                                Team = a.Team.Select(t => new TeamViewModel
+                                {
+                                    ID = t.ID,
+                                    Name = t.Name,
+                                    Status = _context.TeamStatus
+                                                .FirstOrDefault(ts => ts.ID == t.StatusID)
+                                                .Description,
+                                                
+                                    Logo = t.Logo
+                                })
+                            
+                            })
```

Comment:
Use the implicit operator that I have recommended you create.

Reply:
mentioned above, but another issue is that some of fields (for now, status) has a map between ID and description. I'm not sure if I should or even could import _context into a viewmodel.

Reply:
Yea, a helper method would be better.